docs/timeseries/lombscarglemb.rst

```diff
@@ -0,0 +1,413 @@
+...     y_band += dy_band * rng.standard_normal(N)
+...     t += list(t_band)
+...     y += list(y_band)
+...     dy += list(dy_band)
+...     bands += [band] * N
+
+The Lomb-Scargle periodogram, evaluated at frequencies chosen
+automatically based on the input data, can be computed as follows
+using the :class:`~astropy.timeseries.LombScargleMultiband` class,
+with the ``bands`` argument being the sole difference in comparison
+to the :class:`~astropy.timeseries.LombScargle` interface:
+
+>>> from astropy.timeseries import LombScargleMultiband
+>>> frequency,power = LombScargleMultiband(t, y, bands, dy).autopower()
+
+Plotting the result with Matplotlib gives:
+
+.. plot::
+
+   import numpy as np
+   import matplotlib.pyplot as plt
+   from astropy.timeseries import LombScargleMultiband
+
+   t = []
+   y = []
+   bands = []
+   dy = []
+
+   N=60
+
+   for i,band in enumerate(['u','g','r','i','z']):
+      rng = np.random.default_rng(i)
+      t_band = 300 * rng.random(N)
+      y_band = 3 + 2 * np.sin(2 * np.pi * t_band)
+      dy_band = 0.01 * (0.5 + rng.random(N))
+      y_band += dy_band * rng.standard_normal(N)
+
+      t += list(t_band)
+      y += list(y_band)
+      dy += list(dy_band)
+      bands += [band] * N
+
+   frequency,power = LombScargleMultiband(t,y,bands,dy).autopower()
+
+   plt.plot(frequency,power)
+
+The periodogram shows a clear spike at a frequency of 1 cycle per unit time,
+as we would expect from the data we constructed. The resulting power is a
+single array, with combined input from each of the bands dependent upon the
+implementation chosen in the ``method`` keyword.
+
+.. EXAMPLE END
+
+Periodograms from :class:`~astropy.timeseries.TimeSeries` objects
+=================================================================
+
+:class:`~astropy.timeseries.LombScargleMultiband` is able to operate on
+:class:`~astropy.timeseries.TimeSeries` objects, provided the
+:class:`~astropy.timeseries.TimeSeries` object meets a formatting requirement.
+The requirement is that the flux (or magnitudes) and errors for each band are
+provided in separate columns. If instead, your
+:class:`~astropy.timeseries.TimeSeries` object has a singular flux column with
+an associated band label column, these columns may be passed directly to
+:class:`~astropy.timeseries.LombScargleMultiband` as 1-d arrays.
+
+Example
+-------
+
+.. EXAMPLE START: Loading from a :class:`~astropy.timeseries.TimeSeries` object
+
+Consider the following generator code for a
+:class:`~astropy.timeseries.TimeSeries` object. Where timeseries data is
+populated for three photometric bands (g,r,i).
+
+>>> from astropy.timeseries import LombScargleMultiband, TimeSeries
+>>> import numpy as np
+>>> import astropy.units as u
+
+>>> rng = np.random.default_rng(1)
+>>> deltas = 240 * rng.random(180)
+>>> ts1 = TimeSeries(time_start="2011-01-01T00:00:00",
+...                  time_delta=deltas*u.minute)
+
+>>> # g band fluxes
+>>> ts1['g_flux'] = [np.nan] * 180 * u.mJy
```

Comment:
LombScargleMultiband treats NaN values the same way LombScargle does, which is to say that both classes don't handle them at all when passing arrays with NaN values directly through. But both check for them in the y and dy columns when ingest from a timeseries object. As for proper masked values, do you mean like from a numpy.masked.array? Under the hood it's just using np.isnan calls, which should work well with numpy's masked arrays, and from then on we treat them like numpy arrays, which I believe masked arrays naturally accommodate.

Reply:
As long as this is consistent with LombScargle this is fine for now - but yes I mean tables with missing values as shown here:

https://docs.astropy.org/en/stable/table/masking.html

This is the 'preferred' way of indicating missing data as opposed to using NaN values.

Reply:
If you have time, could you check if masked tables work as well as NaN values? If so, I wonder if a masked table would look nicer in the docs? (but ignore for now if short on time).

Reply:
I just updated the from_timeseries function to support proper masked values, and updated the docs to show a masked table instead of just a table with NaN values. Also updated the test suite to use proper masking. So just to be clear, if a user creates a timeseries object and loads it into LombScargleMultiband, it recognizes proper masking and handles it accordingly. The LombScargleMultiband class itself still doesn't handle NaN values if they are included in array inputs, which is how LombScargle behaves. I would think a change to this behavior should go in a separate PR that also modifies LombScargle to keep the two consistent.